src/hooks/__tests__/useLoginApi.test.ts

```diff
@@ -0,0 +1,230 @@
+import { act } from 'react';
+import { renderHook } from '@testing-library/react';
+import axios from 'axios';
+import cookies from 'js-cookie';
+import { useLoginApi, useLogoutApi, useUserApi, useGetSetAuthApi } from '../useLoginApi';
+
+describe('useLoginApi', () => {
+  let spyCookies;
+  let hookResult;
+
+  beforeEach(() => {
+    spyCookies = jest.spyOn(cookies, 'set');
+    const hook = renderHook(() => useLoginApi());
+    hookResult = hook?.result?.current;
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should attempt an api call to login', () => {
+    const { apiCall } = hookResult;
+    const spyAxios = jest.spyOn(axios, 'post');
+
+    apiCall({ username: 'lorem', password: '123456' });
+    expect(spyAxios.mock.calls).toMatchSnapshot('apiCall');
+  });
+
+  it('should handle success while attempting to login a user', async () => {
+    const { login } = hookResult;
+    jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({}));
+
+    await expect(login({ username: 'lorem', password: '123456' })).resolves.toMatchSnapshot('login, success');
+  });
+
+  it('should handle errors while attempting to login a user', async () => {
+    const { login } = hookResult;
+    jest.spyOn(axios, 'post').mockImplementation(() => Promise.reject({ isAxiosError: true, message: 'Mock error' }));
```

Comment:
optional: review relaxing the linting rule on Promise rejection errors. there are scenarios, like this one that don't require it... and doubly so on test files

Reply:
done, all test files will be ignored for that rule